ra2ce/graph/networks.py

```diff
@@ -179,6 +194,9 @@ def network_shp(
         edges_complex = nut.join_nodes_edges(nodes, edges, id_name)
         edges_complex.crs = crs  # set the right CRS
 
+        edges_complex = edges_complex.loc[~edges_complex["node_A"].isnull()]
+        edges_complex = edges_complex.loc[~edges_complex["node_B"].isnull()]
```

Comment:
I believe there was a way to combine both options, I don't remember them though :). Either way, same as earlier, when you are 'reassigning' something is not usually a good sign. Consider extracting it into a separate method such as `get_edges_complex` so we know exactly what to expect once you get the returned object.

Reply:
Fixed for the part 1 of the comment. :). 
I think these two lines were temporary fix to circumvent the two assert rules below. I agree we should improve this but I prefer to do that later together with Frederique to understand better what to expect.